src/main/java/util/miscellaneous/DukeOperation.java

```diff
@@ -0,0 +1,321 @@
+        linePrinter();
+        System.out.println("\t" + MARKED_MSG);
+        System.out.println("\t" + "   " + (tasks.get(markedItem)).toString());
+        linePrinter();
+
+    }
+
+    public static void printUnmark(ArrayList<Task> tasks, int unmarkedItem) {
+        linePrinter();
+        System.out.println("\t" + UNMARKED_MSG);
+        System.out.println("\t" + "   " + (tasks.get(unmarkedItem)).toString());
+        linePrinter();
+
+    }
+
+    public static void checkFilePath() throws IOException {
+        File file = new File(FILEPATH);
+        file.getParentFile().mkdirs();
+        file.createNewFile();
+    }
+
+    public static void saveToFile(ArrayList<Task> tasks) throws IOException {
+        FileWriter writer = new FileWriter(FILEPATH);
+        for (int i = 0; i < tasks.size(); i++) {
+            writer.write((tasks.get(i).toString()));
+            writer.write(System.lineSeparator());
+        }
+        writer.close();
+    }
+
+    public static void saveData(ArrayList<Task> tasks) {
+        try {
+            checkFilePath();
+            saveToFile(tasks);
+        } catch (IOException e) {
+            echo(IO_ERROR_MSG);
+        }
+    }
+
+    public static void loadData(ArrayList<Task> tasks) {
+        try {
+            File file = new File(FILEPATH);
+            Scanner input1 = new Scanner(file);
+            String load;
+
+            while (input1.hasNext()) {
+                load = input1.nextLine();
+                boolean needUpdateTaskStatus = false;
+                boolean isLoadingData = true;
+
+                if(load.indexOf(DONE_STATUS) == TASK_STATUS_INDEX) {
+                    needUpdateTaskStatus = true;
+                }
+
+                if (load.startsWith(DEADLINE_TYPE)) {
+                    String time = load.substring(load.lastIndexOf("(by: "));
+                    load = ADD_DEADLINE_CMD + load.substring(TASK_DETAIL, load.indexOf("by ")) + DEADLINE_OF_TASK_CMD + " " + time;
+                } else if (load.startsWith(EVENT_TYPE)) {
+                    String time = load.substring(load.lastIndexOf("(by: "));
+                    load = ADD_EVENT_CMD + load.substring(TASK_DETAIL, load.indexOf("by ")) + DURATION_OF_EVENT_CMD + " " + time;
+                } else {
+                    load = ADD_TODO_CMD + load.substring(TASK_DETAIL);
+                }
+
+                loadAndRun(tasks, load, isLoadingData, needUpdateTaskStatus);
+            }
+        } catch (FileNotFoundException e) {
+            echo(NO_PREVIOUS_RECORD);
+        }
+    }
+
+    public static void exitLine() {
+        linePrinter();
+        System.out.println("\t" + GOODBYE_MSG);
+        linePrinter();
+    }
+
+    public static void echo (String line) {
+        linePrinter();
+        System.out.println("\t" + " " + line);
+        linePrinter();
+    }
+
+    public static CommandType findCommandType(String line) {
+        CommandType c;
```

Comment:
Instead of naming with `c`, you could consider renaming as a [noun](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#use-nouns-for-things-and-verbs-for-actions) for better code quality.